FileXT/dllmain.cpp

```diff
@@ -0,0 +1,347 @@
+// dllmain.cpp : Defines the entry point for the DLL application.
+#include "pch.h"
+#include "common.h"
+#include "errorCodes.h"
+#include "filemgr.h"
+#include "value.h"
+
+
+#ifndef _MSC_VER
+#include <dlfcn.h>
+#endif
+
+using namespace std;
+
+// Globals
+filext::filemgr gFileMgr;
+
+#ifndef NDEBUG
+FILE* gLogFile = nullptr;
+#endif
+
+bool checkFileName(string& fileName);
+string getDllFolder();
+
+const std::string& GetAndEnsureStorageDir()
```

Comment:
I tried to avoid refactoring, this change was to avoid crashes I couldn't effectively debug without a linux development environment. I do like this change, but strictly speaking it shouldn't be necessary for this PR.